### DISCUSSION TITLE
set_configvar() for a directory path is not correct for backslashes in Windows.

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#2676] [Q&A] [answered]
`set_configvar("DIR", "$(projectdir)")` with `add_configfiles` will generate a macro in `config.h`, but this macro will have only one backslash in C/C++. To prevent escape characters, shouldn't it be a raw string or have two backslashes?

Or is there any way that I can pass the directory macro to my C/C++ program?

**Comment by waruqi** [comment, marked as answer]
`set_configvar("DIR", os.projectdir():gsub("\\", "/"))`

**Reply by Extra-Creativity** [reply]
It seems that `gsub` returns a table `[replacedStr, matchTimes]`, so maybe it needs some simple process. I've changed it in my file :).

Thanks for your tolerance to my ignorance on XMake and lua (XD. I will forge ahead on studying it.

**Reply by waruqi** [reply]
use () to wrap it, `set_configvar("DIR", (os.projectdir():gsub("\\", "/")))`
